viewing_party/party.py

```diff
@@ -1,23 +1,166 @@
+      "genre": genre,
+      "rating": rating,
+    }
+  else:
+    return None
+
+def add_to_watched(user_data, movie):
+  user_data["watched"].append(movie) #append returns None when assigned to a variable
+  return user_data
+
+def add_to_watchlist(user_data, movie): #adding movides to watchlist
+  user_data["watchlist"].append(movie) #via appending movie to dict
+  return user_data
+
+def watch_movie(user_data, title):
+  for i in range(len(user_data["watchlist"])): #used for loop to iterate through n number of watchlists
+    if title == user_data["watchlist"][i]["title"]: #suggestion to make variables
+      user_data["watchlist"].pop(i) #leads to empty watchlist, pop only works for index
+      user_data["watched"].append(title)
+    #print(user_data)
+  return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+  average_rating = 0
+  sum_rating = 0
+  count = 0
+  for i in range(len(user_data["watched"])):
+    rating = user_data["watched"][i]['rating']
+    sum_rating += rating
+    count += 1 #adding counter everytime goes through watched dictionary
+  if count != 0:
+    average_rating = sum_rating / count
+    return average_rating
+  else:
+    return average_rating #this works because avg_rating italized to 0
+
+def get_most_watched_genre(user_data):
+  genres_list = []
+  for i in range(len(user_data["watched"])):
+    genre = user_data["watched"][i]['genre']
+    genres_list.append(genre)
 
+  if len(genres_list) != 0:
+    most_watched_genre = max(genres_list, key = genres_list.count)
+    #using max and count function to count genres and find the most common element
+    return most_watched_genre
+  else: 
+    return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+  movies_friends_watched = []
+  movies_user_watched = []
+  unique_list_user_watched = []
+
+  for i in user_data["watched"]: #iterating through every value
+    movies_user_watched.append(i)
+
+  for i in range(len(user_data["friends"])):
+    for item in user_data["friends"][i]["watched"]: #this loop goes into the 2 watched key inside friends key
+      movies_friends_watched.append(item) #append all movies that friends watched
+
+  for i in movies_user_watched: #comparing elements in movies_user_watched to movies_friends_watched
+    if i not in movies_friends_watched:
+      unique_list_user_watched.append(i)
+  return unique_list_user_watched
+
+def get_friends_unique_watched(user_data): #Test 16 is similar to test 15, instead of testing for 
+  #movies watched by user and not friends, we test for movies watched by friends but not user. 
+
+  movies_friends_watched = [] 
+  movies_user_watched = []
+  unique_list_friends_watched = []
+  no_dup_list = []
+
+  for i in user_data["watched"]:

```

Comment:
While using `i` (usually short for index or iterator) is fine, we can be more descriptive here and in the next few lines! Using more descriptive variables like `user_movie` here and `friend` on L90 and `friend_movie` on L94 would make improve readability!